apps/ml-service/src/main.py

```
@@ -210,11 +211,87 @@ def forecast_drug(
+            "models_reloaded": len(service.models),
+            "timestamp": datetime.now().isoformat()
+        }
+        
+    except ImportError:
+        return {
+            "message": "Training module not found",
+            "note": "Please implement retrain_models_with_recent_data() in models/train.py",
+            "status": "error"
+        }
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Training failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)